src/storage/v3/indices.cpp

```diff
@@ -651,49 +644,63 @@ LabelPropertyIndex::Iterable::Iterator LabelPropertyIndex::Iterable::begin() {
   // If the bounds are set and don't have comparable types we don't yield any
   // items from the index.
   if (!bounds_valid_) {
-    return {this, index_accessor_.end()};
+    return {this, index_accessor_->end()};
   }
-  auto index_iterator = index_accessor_.begin();
+  auto index_iterator = index_accessor_->begin();
   if (lower_bound_) {
-    index_iterator = index_accessor_.find_equal_or_greater(lower_bound_->value());
+    index_iterator = std::ranges::find_if(*index_accessor_, [lower_bound = lower_bound_->value()](const auto &pair) {
+      return pair.value >= lower_bound;
+    });
```

Comment:
Check `std::ranges::lower_bound` [here](https://en.cppreference.com/w/cpp/algorithm/ranges/lower_bound), it might be more efficient (lg vs linear complexity.)

Reply:
Hmm I did not know that, interesting